### DISCUSSION TITLE
latest update crashing Heroku app

### DISCUSSION THREAD

**Discussion by nanacnote** [vercel/next.js#15519] [Help]
# Bug report

## Describe the bug

- I use "latest" as version in my package.json for my project which means whenever I run npm install I get the latest version of next js.

"dependencies": {
    "@types/lodash": "^4.14.157",
    "@types/matter-js": "^0.14.4",
    "bootstrap": "^4.5.0",
    "gsap": "^3.3.4",
    "matter-js": "^0.14.2",
    "next": "latest",  <------------------------------
    "react": "^16.13.1",
    "react-bootstrap": "^1.0.1",
    "react-countup": "^4.3.3",
    "react-dom": "^16.13.1",
    "react-icons": "^3.10.0",
    "sass": "^1.26.9"
  },

- after latest deploy the npm start script is crashing the app on Heroku

"scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start -p $PORT", <------------------------------
    "type-check": "tsc --pretty --noEmit",
    "format": "prettier --write **/*.{js,ts,tsx} **/*/*.{js,ts,tsx}",
    "lint": "eslint . --ext ts --ext tsx --ext js",
    "test": "jest",
    "test-all": "yarn lint && yarn type-check && yarn test"
  },

- when I use a previous next js version, the deploy works fine

"dependencies": {
    "@types/lodash": "^4.14.157",
    "@types/matter-js": "^0.14.4",
    "bootstrap": "^4.5.0",
    "gsap": "^3.3.4",
    "matter-js": "^0.14.2",
    "next": "9.4.5-canary.0",  <------------------------------
    "react": "^16.13.1",
    "react-bootstrap": "^1.0.1",
    "react-countup": "^4.3.3",
    "react-dom": "^16.13.1",
    "react-icons": "^3.10.0",
    "sass": "^1.26.9"
  },

## Heroku Error Log 
H10

**Comment by jamesmosier** [comment]
Do you have any logs you could share?

Also when using `"next": "latest"` if you are generating a `package.lock.json` file then the version will be locked to whatever the version was when you originally installed that dependency.

**Reply by nanacnote** [reply]
Sorry James for late reply I am not getting notified on this thread. Here is a heroku error log it really does not give much away in terms of the exact module that is crashing the app. 

I run npm install before pushing which i think updates the package.lock file?

> 2020-07-30T01:33:45.270614+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
> 2020-07-30T01:33:45.270700+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2020-07-30T01_33_45_266Z-debug.log
> 2020-07-30T01:33:45.315193+00:00 heroku[web.1]: Process exited with status 1
> 2020-07-30T01:33:45.360400+00:00 heroku[web.1]: State changed from starting to crashed
> 2020-07-30T01:33:46.000000+00:00 app[api]: Build succeeded
> 2020-07-30T01:34:40.754445+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=www.adjei-bohyen.me request_id=362863da-53d4-4e33-b603-943b87b61051 fwd="92.40.168.225" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T01:34:43.769851+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/apple-touch-icon-120x120-precomposed.png" host=www.adjei-bohyen.me request_id=78e3d0b1-6377-4bd1-9001-c05598f32dee fwd="92.40.168.225" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T01:34:43.930549+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/apple-touch-icon-120x120.png" host=www.adjei-bohyen.me request_id=43048c72-8062-4343-bcb7-a993e9595800 fwd="92.40.168.225" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T01:34:44.071243+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/apple-touch-icon-precomposed.png" host=www.adjei-bohyen.me request_id=30950352-77e3-4a23-8904-d07bc90af67d fwd="92.40.168.225" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T01:34:44.209948+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/apple-touch-icon.png" host=www.adjei-bohyen.me request_id=ccfc0bd6-73bf-4436-b7fe-dc9dfe348386 fwd="92.40.168.225" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T02:01:02.125096+00:00 heroku[web.1]: State changed from crashed to starting
> 2020-07-30T02:01:05.911162+00:00 heroku[web.1]: Starting process with command `npm start`
> 2020-07-30T02:01:07.917163+00:00 app[web.1]: 
> 2020-07-30T02:01:07.917186+00:00 app[web.1]: > portfolio-app@1.0.0 start /app
> 2020-07-30T02:01:07.917187+00:00 app[web.1]: > next start -p $PORT
> 2020-07-30T02:01:07.917187+00:00 app[web.1]: 
> 2020-07-30T02:01:07.921568+00:00 app[web.1]: sh: 1: next: not found
> 2020-07-30T02:01:07.925066+00:00 app[web.1]: npm ERR! code ELIFECYCLE
> 2020-07-30T02:01:07.925395+00:00 app[web.1]: npm ERR! syscall spawn
> 2020-07-30T02:01:07.925607+00:00 app[web.1]: npm ERR! file sh
> 2020-07-30T02:01:07.925827+00:00 app[web.1]: npm ERR! errno ENOENT
> 2020-07-30T02:01:07.926775+00:00 app[web.1]: npm ERR! portfolio-app@1.0.0 start: `next start -p $PORT`
> 2020-07-30T02:01:07.926894+00:00 app[web.1]: npm ERR! spawn ENOENT
> 2020-07-30T02:01:07.927042+00:00 app[web.1]: npm ERR!
> 2020-07-30T02:01:07.927165+00:00 app[web.1]: npm ERR! Failed at the portfolio-app@1.0.0 start script.
> 2020-07-30T02:01:07.927280+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 2020-07-30T02:01:07.935823+00:00 app[web.1]: 
> 2020-07-30T02:01:07.935980+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
> 2020-07-30T02:01:07.936148+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2020-07-30T02_01_07_928Z-debug.log
> 2020-07-30T02:01:07.995745+00:00 heroku[web.1]: Process exited with status 1
> 2020-07-30T02:01:08.036281+00:00 heroku[web.1]: State changed from starting to crashed
> 2020-07-30T02:37:53.371280+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/_next/static/chunks/df39fdc77665314c6f4844e3ff2cc60981c4f602.3244b981eded637edf7e.js" host=www.adjei-bohyen.me request_id=1e4d2d83-4e0f-4521-805b-0982f70149e3 fwd="66.249.66.89" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T02:37:54.293415+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/_next/static/aLYB-_39IcMy7UFq4zxdb/pages/_app.js" host=www.adjei-bohyen.me request_id=63eb0583-7c83-41b3-ba0b-4d8d5831888c fwd="66.249.66.89" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T02:50:16.883719+00:00 heroku[web.1]: State changed from crashed to starting
> 2020-07-30T02:50:20.542133+00:00 heroku[web.1]: Starting process with command `npm start`
> 2020-07-30T02:50:22.755380+00:00 app[web.1]: 
> 2020-07-30T02:50:22.755401+00:00 app[web.1]: > portfolio-app@1.0.0 start /app
> 2020-07-30T02:50:22.755401+00:00 app[web.1]: > next start -p $PORT
> 2020-07-30T02:50:22.755402+00:00 app[web.1]: 
> 2020-07-30T02:50:22.760152+00:00 app[web.1]: sh: 1: next: not found
> 2020-07-30T02:50:22.764482+00:00 app[web.1]: npm ERR! code ELIFECYCLE
> 2020-07-30T02:50:22.764927+00:00 app[web.1]: npm ERR! syscall spawn
> 2020-07-30T02:50:22.765120+00:00 app[web.1]: npm ERR! file sh
> 2020-07-30T02:50:22.765358+00:00 app[web.1]: npm ERR! errno ENOENT
> 2020-07-30T02:50:22.766783+00:00 app[web.1]: npm ERR! portfolio-app@1.0.0 start: `next start -p $PORT`
> 2020-07-30T02:50:22.766984+00:00 app[web.1]: npm ERR! spawn ENOENT
> 2020-07-30T02:50:22.767201+00:00 app[web.1]: npm ERR!
> 2020-07-30T02:50:22.767432+00:00 app[web.1]: npm ERR! Failed at the portfolio-app@1.0.0 start script.
> 2020-07-30T02:50:22.767648+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 2020-07-30T02:50:22.776019+00:00 app[web.1]: 
> 2020-07-30T02:50:22.776337+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
> 2020-07-30T02:50:22.776534+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2020-07-30T02_50_22_769Z-debug.log
> 2020-07-30T02:50:22.837252+00:00 heroku[web.1]: Process exited with status 1
> 2020-07-30T02:50:22.878305+00:00 heroku[web.1]: State changed from starting to crashed
> 2020-07-30T04:27:32.344139+00:00 heroku[web.1]: State changed from crashed to starting
> 2020-07-30T04:27:35.891173+00:00 heroku[web.1]: Starting process with command `npm start`
> 2020-07-30T04:27:38.256963+00:00 app[web.1]: 
> 2020-07-30T04:27:38.256978+00:00 app[web.1]: > portfolio-app@1.0.0 start /app
> 2020-07-30T04:27:38.256979+00:00 app[web.1]: > next start -p $PORT
> 2020-07-30T04:27:38.256982+00:00 app[web.1]: 
> 2020-07-30T04:27:38.270260+00:00 app[web.1]: sh: 1: next: not found
> 2020-07-30T04:27:38.277895+00:00 app[web.1]: npm ERR! code ELIFECYCLE
> 2020-07-30T04:27:38.278302+00:00 app[web.1]: npm ERR! syscall spawn
> 2020-07-30T04:27:38.278530+00:00 app[web.1]: npm ERR! file sh
> 2020-07-30T04:27:38.278800+00:00 app[web.1]: npm ERR! errno ENOENT
> 2020-07-30T04:27:38.280085+00:00 app[web.1]: npm ERR! portfolio-app@1.0.0 start: `next start -p $PORT`
> 2020-07-30T04:27:38.280257+00:00 app[web.1]: npm ERR! spawn ENOENT
> 2020-07-30T04:27:38.280440+00:00 app[web.1]: npm ERR!
> 2020-07-30T04:27:38.280608+00:00 app[web.1]: npm ERR! Failed at the portfolio-app@1.0.0 start script.
> 2020-07-30T04:27:38.280762+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 2020-07-30T04:27:38.291895+00:00 app[web.1]: 
> 2020-07-30T04:27:38.292201+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
> 2020-07-30T04:27:38.292354+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2020-07-30T04_27_38_285Z-debug.log
> 2020-07-30T04:27:38.375628+00:00 heroku[web.1]: Process exited with status 1
> 2020-07-30T04:27:38.405331+00:00 heroku[web.1]: State changed from starting to crashed
> 2020-07-30T07:14:21.917236+00:00 heroku[web.1]: State changed from crashed to starting
> 2020-07-30T07:14:25.537651+00:00 heroku[web.1]: Starting process with command `npm start`
> 2020-07-30T07:14:27.429538+00:00 app[web.1]: 
> 2020-07-30T07:14:27.429555+00:00 app[web.1]: > portfolio-app@1.0.0 start /app
> 2020-07-30T07:14:27.429556+00:00 app[web.1]: > next start -p $PORT
> 2020-07-30T07:14:27.429556+00:00 app[web.1]: 
> 2020-07-30T07:14:27.433720+00:00 app[web.1]: sh: 1: next: not found
> 2020-07-30T07:14:27.437588+00:00 app[web.1]: npm ERR! code ELIFECYCLE
> 2020-07-30T07:14:27.437904+00:00 app[web.1]: npm ERR! syscall spawn
> 2020-07-30T07:14:27.438038+00:00 app[web.1]: npm ERR! file sh
> 2020-07-30T07:14:27.438198+00:00 app[web.1]: npm ERR! errno ENOENT
> 2020-07-30T07:14:27.439092+00:00 app[web.1]: npm ERR! portfolio-app@1.0.0 start: `next start -p $PORT`
> 2020-07-30T07:14:27.439220+00:00 app[web.1]: npm ERR! spawn ENOENT
> 2020-07-30T07:14:27.439335+00:00 app[web.1]: npm ERR!
> 2020-07-30T07:14:27.439411+00:00 app[web.1]: npm ERR! Failed at the portfolio-app@1.0.0 start script.
> 2020-07-30T07:14:27.439512+00:00 app[web.1]: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 2020-07-30T07:14:27.445647+00:00 app[web.1]: 
> 2020-07-30T07:14:27.445822+00:00 app[web.1]: npm ERR! A complete log of this run can be found in:
> 2020-07-30T07:14:27.445927+00:00 app[web.1]: npm ERR!     /app/.npm/_logs/2020-07-30T07_14_27_440Z-debug.log
> 2020-07-30T07:14:27.488600+00:00 heroku[web.1]: Process exited with status 1
> 2020-07-30T07:14:27.524858+00:00 heroku[web.1]: State changed from starting to crashed
> 2020-07-30T09:50:29.850633+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=www.adjei-bohyen.me request_id=2d396785-5b34-4e52-bec7-e877f5dbbef4 fwd="92.40.193.122" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T09:51:09.558199+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=www.adjei-bohyen.me request_id=acb10d08-c8fd-42c4-8d10-65d0bd3c2bdd fwd="92.40.193.122" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T09:56:54.878420+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=owusu-portfolio.herokuapp.com request_id=b9137dab-a647-4b7e-9188-ee9444c3fd67 fwd="92.40.193.123" dyno= connect= service= status=503 bytes= protocol=http
> 2020-07-30T10:08:51.397196+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=www.adjei-bohyen.me request_id=d5223b0f-dc94-46ee-8490-a04303db8bcc fwd="92.40.193.123" dyno= connect= service= status=503 bytes= protocol=http
>

**Reply by justincy** [reply]
It's saying that the commend `next` is not found. In other words, NPM isn't able to find the `next` script in the `node_modules/.bin` directory. That's really odd. I also deploy to Heroku but I haven't had any trouble with this.

**Reply by justincy** [reply]
However, I run next a little a differently. I have a custom server. So I guess my success is irrelevant.

**Comment by nanacnote** [comment]
at moment I have updated my package.json to look like this with an older next update and it works : |

>   "dependencies": {
    "@types/howler": "^2.2.1",
    "@types/jquery": "^3.5.1",
    "@types/lodash": "^4.14.157",
    "@types/matter-js": "^0.14.4",
    "bootstrap": "^4.5.0",
    "gsap": "^3.3.4",
    "howler": "^2.2.0",
    "jquery": "^3.5.1",
    "matter-js": "^0.14.2",
    "next": "^9.4.4", <------------------------------
    "react": "^16.13.1",
    "react-bootstrap": "^1.0.1",
    "react-countup": "^4.3.3",
    "react-dom": "^16.13.1",
    "react-icons": "^3.10.0",
    "sass": "^1.26.9"
  },

**Reply by jamesmosier** [reply]
Could you try `"next": "9.5.1"` and then run `npm install` which should then update your `package.lock.json`?

**Reply by nanacnote** [reply]
@jamesmosier  that crushes the app on build with Heroku
